src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -3119,11 +3124,100 @@ def generate(cls, config: SubsetConfig) -> Compose:
+            if (input_size := getattr(config, "input_size", None)) is not None:
+                cls._eval_input_size(cfg_transform, input_size)
             transform = cls._dispatch_transform(cfg_transform)
             transforms.append(transform)
 
         return Compose(transforms)
 
+    @classmethod
+    def _eval_input_size(cls, cfg_transform: dict[str, Any], input_size: int | tuple[int, int] | None) -> None:
+        """Evaluate the input_size and replace the placeholder in the init_args.
+
+        Input size should be specified as $(input_size). (e.g. $(input_size) * 0.5)
+        Only simple multiplication or division evaluation is supported. For example,
+        $(input_size) * -0.5    => supported
+        $(input_size) * 2.1 / 3 => supported
+        $(input_size) + 1       => not supported
+        The function decides to pass tuple type or int type based on the type hint of the argument.
+        float point values are rounded to int.
+        """
+        if input_size is None:
+            return
+        _input_size: tuple[int, int] = (input_size, input_size) if isinstance(input_size, int) else tuple(input_size)  # type: ignore[assignment]
+
+        def check_type(value: Any, expected_type: Any) -> bool:  # noqa: ANN401
+            try:
+                typeguard.check_type(value, expected_type)
+            except typeguard.TypeCheckError:
+                return False
+            return True
+
+        model_cls = None
+        for key, val in cfg_transform["init_args"].items():
+            if not (isinstance(val, str) and "$(input_size)" in val):
+                continue
+            if model_cls is None:
+                model_cls = get_obj_from_str(cfg_transform["class_path"])
+
+            available_types = typing.get_type_hints(model_cls.__init__).get(key)
+            if available_types is None or check_type(_input_size, available_types):  # pass tuple[int, int]
+                cfg_transform["init_args"][key] = cls._safe_eval(
+                    val.replace("$(input_size)", f"({','.join(str(val) for val in _input_size)})"),
+                )
+            elif check_type(_input_size[0], available_types):  # pass int
+                cfg_transform["init_args"][key] = cls._safe_eval(val.replace("$(input_size)", str(_input_size[0])))
+            else:
+                msg = f"{key} argument should be able to get int or tuple[int, int], but it can get {available_types}"
+                raise RuntimeError(msg)
+
+    @classmethod
+    def _safe_eval(cls, str_to_eval: str) -> tuple[int, ...] | int:
+        """Safe eval function for _eval_input_size.
+
+        The function is implemented for `_eval_input_size`, so implementation is aligned to it as below
+        - Only multiplication or division evaluation are supported.
+        - Only constant and tuple can be operand.
+        - tuple is changed to numpy array before evaluation.
+        - result value is rounded to int.
+        """
```

Comment:
To me is looks like doing actually `_eval_input_size_str()`.

Reply:
yes, I think it's better to change function name. Thanks :)

Reply:
done https://github.com/openvinotoolkit/training_extensions/pull/3759/commits/589c4c57264a27a5da2cfd9bb9aa3b2788c53bef